src/expr.rs

```diff
@@ -1697,21 +1697,23 @@ fn rewrite_call_inner<R>(context: &RewriteContext,
     // we format the function arguments horizontally.
     let overflow_last = match args.last().map(|x| &x.node) {
         Some(&ast::ExprKind::Closure(..)) |
-        Some(&ast::ExprKind::Block(..)) if arg_count > 1 => true,
+        Some(&ast::ExprKind::Block(..)) => true,
         _ => false,
     };
 
     let mut orig_last = None;
     let mut placeholder = None;
 
+    fn max_line_width(s: &str) -> usize {
+        s.lines()
+            .fold(0,
+                  |max, line| if line.len() > max { line.len() } else { max })
+    }
+
     // Replace the last item with its first line to see if it fits with
     // first arguments.
-    if overflow_last {
-        let nested_shape = Shape {
-            indent: nested_shape.indent.block_only(),
-            ..nested_shape
-        };
-        let rewrite = args.last().unwrap().rewrite(context, nested_shape);
+    if overflow_last && arg_count > 1 {
+        let rewrite = args.last().unwrap().rewrite(context, nested_shape.block());
```

Comment:
I think this change (block_only to block) may be what causes the regression in the last commit